From 8784e9e4f5d6d309fad019cb5942abbff6f046b2 Mon Sep 17 00:00:00 2001
From: Edward Wu <edwardwu@realtek.com>
Date: Mon, 26 Dec 2022 07:51:19 +0800
Subject: [PATCH 1/2] r8169soc: Use a 400-us flush timer to improve CPU loading
 on Android.

jira: PERF-825
CPU loading is an important resource during video streaming.
By 400us GRO flush timer, improves GRO aggregation from
1 ~ 2 pkts to 7 ~ 12 pkts on Youtube video streaming.

This change reduces CPU avg loading about 12% by 400us.

And no different latency results on netperf and ping.

Change-Id: I4e21d68dc1d652b452b8e36f2b2b36a22652e36e
Signed-off-by: Edward Wu <edwardwu@realtek.com>
(cherry picked from commit 7f43411dad482a3118895f2b9cf545de5618acba)
---
 drivers/net/ethernet/realtek/r8169soc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/realtek/r8169soc.c b/drivers/net/ethernet/realtek/r8169soc.c
index 9a869d8f3596..d5910a2bf357 100644
--- a/drivers/net/ethernet/realtek/r8169soc.c
+++ b/drivers/net/ethernet/realtek/r8169soc.c
@@ -11772,6 +11772,8 @@ static int rtl_init_one(struct platform_device *pdev)
 	ndev->hw_features |= NETIF_F_RXALL;
 	ndev->hw_features |= NETIF_F_RXFCS;
 
+	ndev->gro_flush_timeout = 400000;
+
 	ndev->min_mtu = ETH_ZLEN;
 	ndev->max_mtu = tp->chip->jumbo_max;
 
-- 
2.45.2

